e2e-tests/portal/cypress/e2e/journeys/dashboard/dashboard-facilities/dashboard-facilities-filter-by-product.spec.js

```diff
@@ -105,32 +126,52 @@ context('Dashboard Facilities filters - filter by product/facility type', () =>
       cy.url().should('eq', relative('/dashboard/facilities/0'));
     });
 
-    it('submits the filter and redirects to the dashboard', () => {
+    beforeEach(() => {
+      cy.saveSession();
+      dashboardFacilities.visit();
+
       // toggle to show filters (hidden by default)
       filters.showHideButton().click();
+    });
 
+    it('submits the filter and redirects to the dashboard', () => {
       submitRedirectsToDashboard(dashboardFacilities.filters.panel.form.type.contingent.checkbox());
     });
 
     it('renders checked checkbox', () => {
-      // toggle to show filters (hidden by default)
-      filters.showHideButton().click();
-
+      submitRedirectsToDashboard(dashboardFacilities.filters.panel.form.type.contingent.checkbox());
       shouldRenderCheckedCheckbox(dashboardFacilities.filters.panel.form.type.contingent.checkbox());
     });
 
     it('renders the applied filter in the `applied filters` section', () => {
+      submitRedirectsToDashboard(dashboardFacilities.filters.panel.form.type.contingent.checkbox());
+
+      filters.showHideButton().click();
+
       shouldRenderAppliedFilterInPanelSelectedFilters('Product', CONSTANTS.FACILITY.FACILITY_TYPE.CONTINGENT);
     });
 
     it('renders the applied filter in the `main container selected filters` section', () => {
+      submitRedirectsToDashboard(dashboardFacilities.filters.panel.form.type.contingent.checkbox());
+
+      filters.showHideButton().click();
+
       shouldRenderAppliedFilterInMainContainerSelectedFilters(
         dashboardFacilities.filters.mainContainer.selectedFilters.typeContingent(),
         CONSTANTS.FACILITY.FACILITY_TYPE.CONTINGENT,
       );
     });
 
     it('renders only Contingent facilities', () => {
+      submitRedirectsToDashboard(dashboardFacilities.filters.panel.form.type.contingent.checkbox());
+
+      filters.showHideButton().click();
```

Comment:
These 2 lines are duplicated a lot in this file, is it possible to have in the `beforeEach` or is this required due to cypress upgrade?

Reply:
They are, unfortunately not all tests cases are calling it.

Reply:
`filters.showHideButton().click();` appears in beforeEach()